### DISCUSSION TITLE
Consumer Offset Timestamp Source

### DISCUSSION THREAD

**Discussion by ptoews** [rabbitmq/rabbitmq-server#11757] [Questions] [answered]
Hi,

what is the time source for the timestamps that is used when a consumer offset is provided as a timestamp? Is it when the message was received by the broker?

In my case I need to attach at a stream based on a timestamp that is referring to the creation of the data, which I am setting as a property when publishing. However, RabbitMQ does not use this timestamp when using a time offset, and attaches to the beginning or end of the stream (I tried using artificial timestamps much smaller or larger than current time). Is there any other way to do this?

Thanks in advance for any help.

Edit: I am using python and tried this with both the pika and rstream library.

**Comment by nadavpld** [comment, marked as answer]
+1
(Although using node.js)
Posted a [question](https://groups.google.com/g/rabbitmq-users/c/VITbYlITJKA/m/ZXvvuvdvBgAJ?utm_medium=email&utm_source=footer) in rabbitmq-users google group
